src/main-process/atom-application.coffee

```diff
@@ -33,11 +34,15 @@ class AtomApplication
   # Public: The entry point into the Atom application.
   @open: (options) ->
     unless options.socketPath?
+      username = if process.platform is 'win32' then process.env.USERNAME else process.env.USER
+      atomHomeUnique = path.resolve(process.env.ATOM_HOME)
```

Comment:
One issue here is that case-preserving case-insensitive systems (Windows NTFS, macOS X HFS+ default mode) are going to spawn two instances of Atom for the same folder should the case differ.

Reply:
I don't believe there's any reliable way of detecting whether a filesystem is case-sensitive or not. We could lowercase the whole string before hashing and risk the user not getting an instance on a case-sensitive filesystem. I do think that the likelihood of a user having two identical paths, except case, on a case-sensitive system is lower than seeing arbitrary case differences on the other type, so it would make sense on those grounds.